buildSrc/src/main/kotlin/java-repackage-jars.gradle.kts

```diff
@@ -2,15 +2,16 @@ import java.util.jar.JarEntry
 import java.util.jar.JarFile
 import java.util.jar.JarOutputStream
 import org.gradle.api.internal.file.archive.ZipCopyAction
+import java.nio.file.Files
 
 // This registers a `doLast` action to rewrite the timestamps of the project's output JAR
 afterEvaluate {
 	val jarTask = (tasks.findByName("shadowJar") ?: tasks["jar"]) as Jar
 
 	jarTask.doLast {
 
-		val newFile = createTempFile("rewrite-timestamp")
-		val originalOutput = jarTask.archiveFile.get().getAsFile()
+		val newFile = Files.createTempFile("rewrite-timestamp", null).toFile()
+		val originalOutput = jarTask.archiveFile.get().asFile
```

Comment:
I'm curious, what warning does it report if this change isn't made?

Reply:
`createTempFile` is deprecated: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/create-temp-file.html